In [2]:
import csv
a=open("guns.csv","r")
data=list(csv.reader(a))
print(data[0:5])

[['', 'year', 'month', 'intent', 'police', 'sex', 'age', 'race', 'hispanic', 'place', 'education'], ['1', '2012', '01', 'Suicide', '0', 'M', '34', 'Asian/Pacific Islander', '100', 'Home', '4'], ['2', '2012', '01', 'Suicide', '0', 'F', '21', 'White', '100', 'Street', '3'], ['3', '2012', '01', 'Suicide', '0', 'M', '60', 'White', '100', 'Other specified', '4'], ['4', '2012', '02', 'Suicide', '0', 'M', '64', 'White', '100', 'Home', '4']]


In [3]:
headers=data[0]
data=data[1:]
print(headers)
print(data[0:5])

['', 'year', 'month', 'intent', 'police', 'sex', 'age', 'race', 'hispanic', 'place', 'education']
[['1', '2012', '01', 'Suicide', '0', 'M', '34', 'Asian/Pacific Islander', '100', 'Home', '4'], ['2', '2012', '01', 'Suicide', '0', 'F', '21', 'White', '100', 'Street', '3'], ['3', '2012', '01', 'Suicide', '0', 'M', '60', 'White', '100', 'Other specified', '4'], ['4', '2012', '02', 'Suicide', '0', 'M', '64', 'White', '100', 'Home', '4'], ['5', '2012', '02', 'Suicide', '0', 'M', '31', 'White', '100', 'Other specified', '2']]


In [4]:
years=[row[1] for row in data]
year_counts={}
for year in years:
    if year in year_counts:
        year_counts[year]=year_counts[year]+1
    else:
        year_counts[year]=1
print(year_counts)
    

{'2012': 33563, '2013': 33636, '2014': 33599}


In [5]:
import datetime
dates=[str(datetime.datetime(year=int(row[1]),month=int(row[2]),day=1)) for row in data]
print(dates[0:5])
date_counts={}
for date in dates:
    if date in date_counts:
        date_counts[date]=date_counts[date]+1
    else:
        date_counts[date]=1
print(date_counts)

['2012-01-01 00:00:00', '2012-01-01 00:00:00', '2012-01-01 00:00:00', '2012-02-01 00:00:00', '2012-02-01 00:00:00']
{'2012-01-01 00:00:00': 2758, '2012-02-01 00:00:00': 2357, '2012-03-01 00:00:00': 2743, '2012-04-01 00:00:00': 2795, '2012-05-01 00:00:00': 2999, '2012-06-01 00:00:00': 2826, '2012-07-01 00:00:00': 3026, '2012-08-01 00:00:00': 2954, '2012-09-01 00:00:00': 2852, '2012-10-01 00:00:00': 2733, '2012-11-01 00:00:00': 2729, '2012-12-01 00:00:00': 2791, '2013-01-01 00:00:00': 2864, '2013-02-01 00:00:00': 2375, '2013-03-01 00:00:00': 2862, '2013-04-01 00:00:00': 2798, '2013-05-01 00:00:00': 2806, '2013-06-01 00:00:00': 2920, '2013-07-01 00:00:00': 3079, '2013-08-01 00:00:00': 2859, '2013-09-01 00:00:00': 2742, '2013-10-01 00:00:00': 2808, '2013-11-01 00:00:00': 2758, '2013-12-01 00:00:00': 2765, '2014-01-01 00:00:00': 2651, '2014-02-01 00:00:00': 2361, '2014-03-01 00:00:00': 2684, '2014-04-01 00:00:00': 2862, '2014-05-01 00:00:00': 2864, '2014-06-01 00:00:00': 2931, '2014-07-01 0

In [6]:
data[0:5]

[['1',
  '2012',
  '01',
  'Suicide',
  '0',
  'M',
  '34',
  'Asian/Pacific Islander',
  '100',
  'Home',
  '4'],
 ['2', '2012', '01', 'Suicide', '0', 'F', '21', 'White', '100', 'Street', '3'],
 ['3',
  '2012',
  '01',
  'Suicide',
  '0',
  'M',
  '60',
  'White',
  '100',
  'Other specified',
  '4'],
 ['4', '2012', '02', 'Suicide', '0', 'M', '64', 'White', '100', 'Home', '4'],
 ['5',
  '2012',
  '02',
  'Suicide',
  '0',
  'M',
  '31',
  'White',
  '100',
  'Other specified',
  '2']]

In [7]:
sex_counts={}
race_counts={}
for row in data:
    if row[5] in sex_counts:
        sex_counts[row[5]]=sex_counts[row[5]]+1
    else:
        sex_counts[row[5]]=1
    if row[7] in race_counts:
        race_counts[row[7]]=race_counts[row[7]]+1
    else:
        race_counts[row[7]]=1
print(sex_counts)
print(race_counts)

{'M': 86349, 'F': 14449}
{'Asian/Pacific Islander': 1326, 'White': 66237, 'Native American/Native Alaskan': 917, 'Black': 23296, 'Hispanic': 9022}


## Exploring gun deaths in the US
The annual death rates from gun shots between 2012 and 2014 look relatively stable. The same stands if someone looks at the monthly rates. 

The majority of the victims are male. Looking at race characteristics most of the victims are white people (66237) followed by black people (23296). 

It will be interesting to examine other demographics like age or level of education of the victims, as well as the motives of the gun shots and the involvement of the police. 

In [8]:
import csv
a=open("census.csv","r")
census=list(csv.reader(a))
print(census)

[['Id', 'Year', 'Id', 'Sex', 'Id', 'Hispanic Origin', 'Id', 'Id2', 'Geography', 'Total', 'Race Alone - White', 'Race Alone - Hispanic', 'Race Alone - Black or African American', 'Race Alone - American Indian and Alaska Native', 'Race Alone - Asian', 'Race Alone - Native Hawaiian and Other Pacific Islander', 'Two or More Races'], ['cen42010', 'April 1, 2010 Census', 'totsex', 'Both Sexes', 'tothisp', 'Total', '0100000US', '', 'United States', '308745538', '197318956', '44618105', '40250635', '3739506', '15159516', '674625', '6984195']]


In [9]:
import csv
a=open("guns.csv","r")
data=list(csv.reader(a))
print(data[0:5])
headers=data[0]
data=data[1:]

[['', 'year', 'month', 'intent', 'police', 'sex', 'age', 'race', 'hispanic', 'place', 'education'], ['1', '2012', '01', 'Suicide', '0', 'M', '34', 'Asian/Pacific Islander', '100', 'Home', '4'], ['2', '2012', '01', 'Suicide', '0', 'F', '21', 'White', '100', 'Street', '3'], ['3', '2012', '01', 'Suicide', '0', 'M', '60', 'White', '100', 'Other specified', '4'], ['4', '2012', '02', 'Suicide', '0', 'M', '64', 'White', '100', 'Home', '4']]


In [10]:
sex_counts={}
race_counts={}
for row in data:
    if row[5] in sex_counts:
        sex_counts[row[5]]=sex_counts[row[5]]+1
    else:
        sex_counts[row[5]]=1
    if row[7] in race_counts:
        race_counts[row[7]]=race_counts[row[7]]+1
    else:
        race_counts[row[7]]=1
print(sex_counts)
print(race_counts)

{'M': 86349, 'F': 14449}
{'Asian/Pacific Islander': 1326, 'White': 66237, 'Native American/Native Alaskan': 917, 'Black': 23296, 'Hispanic': 9022}


In [11]:
import csv
a=open("census.csv","r")
census=list(csv.reader(a))
print(census)

[['Id', 'Year', 'Id', 'Sex', 'Id', 'Hispanic Origin', 'Id', 'Id2', 'Geography', 'Total', 'Race Alone - White', 'Race Alone - Hispanic', 'Race Alone - Black or African American', 'Race Alone - American Indian and Alaska Native', 'Race Alone - Asian', 'Race Alone - Native Hawaiian and Other Pacific Islander', 'Two or More Races'], ['cen42010', 'April 1, 2010 Census', 'totsex', 'Both Sexes', 'tothisp', 'Total', '0100000US', '', 'United States', '308745538', '197318956', '44618105', '40250635', '3739506', '15159516', '674625', '6984195']]


In [12]:
census[1]

['cen42010',
 'April 1, 2010 Census',
 'totsex',
 'Both Sexes',
 'tothisp',
 'Total',
 '0100000US',
 '',
 'United States',
 '308745538',
 '197318956',
 '44618105',
 '40250635',
 '3739506',
 '15159516',
 '674625',
 '6984195']

In [13]:
census=census[1]

In [14]:
print(census)

['cen42010', 'April 1, 2010 Census', 'totsex', 'Both Sexes', 'tothisp', 'Total', '0100000US', '', 'United States', '308745538', '197318956', '44618105', '40250635', '3739506', '15159516', '674625', '6984195']


In [15]:
import csv
a=open("census.csv","r")
census=list(csv.reader(a))
print(census)

[['Id', 'Year', 'Id', 'Sex', 'Id', 'Hispanic Origin', 'Id', 'Id2', 'Geography', 'Total', 'Race Alone - White', 'Race Alone - Hispanic', 'Race Alone - Black or African American', 'Race Alone - American Indian and Alaska Native', 'Race Alone - Asian', 'Race Alone - Native Hawaiian and Other Pacific Islander', 'Two or More Races'], ['cen42010', 'April 1, 2010 Census', 'totsex', 'Both Sexes', 'tothisp', 'Total', '0100000US', '', 'United States', '308745538', '197318956', '44618105', '40250635', '3739506', '15159516', '674625', '6984195']]


In [16]:
print(race_counts)

{'Asian/Pacific Islander': 1326, 'White': 66237, 'Native American/Native Alaskan': 917, 'Black': 23296, 'Hispanic': 9022}


In [17]:
mapping={}
census_headers=census[0]
census=census[1]
mapping["Asian/Pacific Islander"]=int(census[15])+int(census[16])
mapping["Black"]=int(census[13])
mapping["Hispanic"]=int(census[12])
mapping["Native American/Native Alaskan"]=int(census[14])
mapping["White"]=int(census[11])

race_per_hundredk={}
for key,value in race_counts.items():
    result=(value/mapping[key])*100000
    race_per_hundredk[key]=result
print(race_per_hundredk)

{'Asian/Pacific Islander': 17.313372033811998, 'White': 148.45318957405297, 'Native American/Native Alaskan': 6.049005786200562, 'Black': 622.9699858751396, 'Hispanic': 22.41455321139654}


In [18]:
intents=[row[3] for row in data]
races=[row[7] for row in data]
homicide_race_counts={}
for i,race in enumerate(races):
    if intents[i]=="Homicide":
        if race in homicide_race_counts:
            homicide_race_counts[race]=homicide_race_counts[race]+1
        else:
            homicide_race_counts[race]=1

for key,value in homicide_race_counts.items():
    result=(value/mapping[key])*100000
    homicide_race_counts[key]=result
print(homicide_race_counts)


{'White': 20.50064654247418, 'Asian/Pacific Islander': 7.298774484842313, 'Black': 521.7266665703972, 'Native American/Native Alaskan': 2.1504644343526533, 'Hispanic': 13.997294701064964}


## Gun Deaths and Homicides for 100k people
Comparing the dataset with census data we found that the most affected group from gun deaths in the US is black people. 

The results, even though reduced, is similar for homicides. However, it's interesting to see that the majority of gun shots where the victims were white people was not related to homicide incidents. 

The next step of the analysis is to examine how these numbers break down for different months of the year

In [19]:
months=[row[2] for row in data]
homicide_month_counts={}
for i,month in enumerate(months):
    if intents[i]=="Homicide":
        if month in homicide_month_counts:
            homicide_month_counts[month]=homicide_month_counts[month]+1
        else:
            homicide_month_counts[month]=1
        

print(homicide_month_counts)
    

{'03': 2780, '04': 2845, '06': 3130, '07': 3269, '08': 3125, '09': 2966, '10': 2968, '12': 3191, '01': 2829, '02': 2178, '05': 2976, '11': 2919}


## Homicides by month
As we can see, the homicides are increased for all summer months as well as for December. Let's explore how the homicide rate varies for gender


In [20]:
gender=[row[5] for row in data]
homicide_gender_counts={}
for i,gen in enumerate(gender):
    if intents[i]=="Homicide":
        if gen in homicide_gender_counts:
            homicide_gender_counts[gen]=homicide_gender_counts[gen]+1
        else:
            homicide_gender_counts[gen]=1
        

print(homicide_gender_counts)

{'M': 29803, 'F': 5373}


## Homicides by gender 
According to the results the homicides with male victims are 5 to 6 times higher than homicides with female victims. However, to calculate the rates for the US population we would need census information on the gender proportions. Now let's explore other types of gun deaths by demographic segments. 

In [21]:
accident_gender_counts={}
for i,gen in enumerate(gender):
    if intents[i]=="Accidental":
        if gen in accident_gender_counts:
            accident_gender_counts[gen]=accident_gender_counts[gen]+1
        else:
            accident_gender_counts[gen]=1
        

print(accident_gender_counts)

{'M': 1421, 'F': 218}


In [22]:
accident_race_counts={}
for i,race in enumerate(races):
    if intents[i]=="Accidental":
        if race in accident_race_counts:
            accident_race_counts[race]=accident_race_counts[race]+1
        else:
            accident_race_counts[race]=1

for key,value in accident_race_counts.items():
    result=(value/mapping[key])*100000
    accident_race_counts[key]=result
print(accident_race_counts)

{'White': 2.5370866826370144, 'Black': 8.771212026401349, 'Hispanic': 0.360242763871924, 'Asian/Pacific Islander': 0.15668209985350223, 'Native American/Native Alaskan': 0.14512336673545514}


## Accidents by race and gender
Again, gun deaths caused by accidents are much higher for men than for women. Compare to homicides, nevertheless, the absolute values are significantly lower. The same applies for the rate of accidents among 100K people. This time the proportion of accidents between black and white people (8.77/2.53) is less emphatic than for homicides (521.73/20.5).

As a final step we will explore how gun death rates correlate with location and education.

In [23]:
location_counts={}
education_counts={}
for row in data:
    if row[9] in location_counts:
        location_counts[row[9]]=location_counts[row[9]]+1
    else:
        location_counts[row[9]]=1
    if row[10] in education_counts:
        education_counts[row[10]]=education_counts[row[10]]+1
    else:
        education_counts[row[10]]=1
print(location_counts)
print(education_counts)

{'Home': 60486, 'Street': 11151, 'Other specified': 13751, 'NA': 1384, 'Other unspecified': 8867, 'Trade/service area': 3439, 'Farm': 470, 'Industrial/construction': 248, 'School/instiution': 671, 'Sports': 128, 'Residential institution': 203}
{'4': 12946, '3': 21680, '2': 42927, '1': 21823, '5': 1369, 'NA': 53}


## Gun deaths by location and education
Breaking the gun deaths in the US by "place of incident", we can see that the majority of these events happen at home. The second largest category includes places that are specified but are not in the above list. Finally, a significant amount of incidents take place on the street. 

The analysis for education shows that the majority of gun deaths happened for high school graduates, while the numbers decrease for higher levels of education. Of course, this needs to be assessed for the ratios of education levels in the population, as we did for the race analysis. 